api/grpcserver/interface.go

```diff
@@ -0,0 +1,9 @@
+package grpcserver
+
+import "context"
+
+//go:generate mockgen -package=grpcserver -destination=./mocks.go -source=./interface.go
+
+type txValidator interface {
```

Comment:
other interfaces are defined right in the api/ . maybe that wasn't necessary, but maybe for consistency worth moving this one as well

Reply:
I can move it there for now.

Imo in a future PR we can merge the packages `api`, `api/config` and `api/grpcserver` into a single package `grpcserver`: The interfaces defined in the `api` package are only used in the `api/grpcserver` package, while their name - e.g. `api.Syncer` - would suggest they are intended to be used in multiple places. The `api/config` package contains configuration for the `api/grpcserver` package. `config.DefaultConfig()` is less expressive than `grpcserver.DefaultConfig()` and also has the disadvantage that we have to use named imports whenever we use more than one "config" package.

What do you think?

Reply:
sounds good

Reply:
created PR: https://github.com/spacemeshos/go-spacemesh/pull/4297